pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test.TestDataDriven [failed](https://mesolite.cluster.engflow.com/invocations/default/e4780f4e-e56d-4e0c-92a3-35ee70137c7a?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NwYW5jb25maWdjY2wvc3BhbmNvbmZpZ2xpbWl0ZXJjY2w6c3BhbmNvbmZpZ2xpbWl0ZXJjY2xfdGVzdA==) on master @ [daec8eb58cf82768a4e2c6336324e4779d857a6e](https://github.com/cockroachdb/cockroach/commits/daec8eb58cf82768a4e2c6336324e4779d857a6e):


```
=== RUN   TestDataDriven
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestDataDriven871672249
    test_log_scope.go:82: use -show-logs to present logs inline
    datadriven_test.go:150: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven871672249
--- FAIL: TestDataDriven (415.26s)
=== RUN   TestDataDriven/partitioning
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:5: still running after 10.014188897s
    datadriven_test.go:87: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:5:
          initialize tenant=11
          ----
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 10.011857958s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 20.012832352s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 30.021530927s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 40.027904095s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 50.031424726s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 1m0.051348508s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 1m10.053691873s
    datadriven.go:357: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8: still running after 1m20.06291055s
    datadriven_test.go:87: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfiglimiterccl/spanconfiglimiterccl_test_/spanconfiglimiterccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfiglimiterccl/testdata/partitioning:8:
          exec-sql tenant=11
          CREATE DATABASE db;
          CREATE TABLE db.list_partitions(i INT PRIMARY KEY, j INT, k INT);
          CREATE INDEX idx_i ON db.list_partitions (i);
          CREATE INDEX idx_j ON db.list_partitions (j);
          CREATE INDEX idx_k ON db.list_partitions (k);
          ALTER INDEX db.list_partitions@idx_i PARTITION BY LIST (i) (
            PARTITION one_and_five    VALUES IN (1, 5),
            PARTITION everything_else VALUES IN (DEFAULT)
          );
        
        expected:
        
        found:
          err: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=71b04e55 key=/Tenant/11/Table/3/1/106/2/1 iso=Serializable pri=0.02364973 epo=0 ts=1777288215.255362248,2 min=1777288209.982302260,0 seq=9} lock=true stat=PENDING rts=1777288209.982302260,0 gul=1777288210.482302260,0 obs={n1@1777288210.287117435,0}
--- FAIL: TestDataDriven/partitioning (113.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63279

--- a/pkg/ccl/spanconfigccl/spanconfiglimiterccl/datadriven_test.go
+++ b/pkg/ccl/spanconfigccl/spanconfiglimiterccl/datadriven_test.go
@@ -49,6 +49,7 @@ import (
 func TestDataDriven(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderRace(t, "too slow under race")
 
 	ctx := context.Background()
 	datadriven.Walk(t, datapathutils.TestDataPath(t), func(t *testing.T, path string) {

